src/transactions/saga.ts

```diff
@@ -225,7 +234,60 @@ function* watchAddressToE164PhoneNumberUpdate() {
   )
 }
 
+function* getTransactionReceipt(
+  transaction: StandbyTransaction & { transactionHash: string },
+  network: Network
+) {
+  const { transactionHash } = transaction
+
+  try {
+    const receipt = yield* call(publicClient[network].getTransactionReceipt, {
+      hash: transactionHash as Hash,
+    })
+
+    if (receipt) {
+      yield* put(
+        transactionConfirmed(transaction.context.id, {
+          transactionHash: receipt.transactionHash,
+          block: receipt.blockNumber.toString(),
+          status: receipt.status == 'success',
+        })
+      )
+    }
+  } catch (e) {
+    Logger.warn(
+      TAG,
+      `Error found when trying to fetch status for transaction with hash: ${transactionHash} in ${network}`,
+      e
+    )
+    return
+  }
+}
+
+export function* watchPendingTransactionsInNetwork(network: Network) {
+  while (true) {
+    const pendingStandbyTransactions = yield* select(pendingStandbyTransactionsSelector)
+    const filteredPendingTxs = pendingStandbyTransactions.filter((tx) => {
+      return tx.networkId === networkConfig.networkToNetworkId[network] && tx.transactionHash
+    })
+
+    for (const transaction of filteredPendingTxs) {
+      yield* fork(getTransactionReceipt, transaction, network)
+    }
+
+    yield* delay(WATCHING_DELAY_BY_NETWORK[network])
+  }
+}
+
+export function* watchPendingTransactions() {
+  const supportedNetworks = Object.keys(publicClient) as Network[]
+  for (const network of supportedNetworks) {
```

Comment:
could we reduce the number of layers / spawned sagas? watchPendingTransactions could handle all networks (like it can select all pending tx's that have a hash, and for each, spawn the `getTransactionReceipt` saga with the network fromthe transaction itself?

Reply:
Hmm, if we want to have a delay configurable by chain, we need the flows to be separated, right?
What's the problem with creating a saga per chain for this?